src/librustc_mir/mir_map.rs

```diff
@@ -227,15 +225,16 @@ fn closure_self_ty<'a, 'tcx>(tcx: &ty::ctxt<'tcx>,
     let region = tcx.mk_region(region);
 
     match tcx.closure_kind(tcx.map.local_def_id(closure_expr_id)) {
-        ty::ClosureKind::FnClosureKind =>
-            tcx.mk_ref(region,
-                       ty::TypeAndMut { ty: closure_ty,
-                                        mutbl: hir::MutImmutable }),
-        ty::ClosureKind::FnMutClosureKind =>
-            tcx.mk_ref(region,
-                       ty::TypeAndMut { ty: closure_ty,
-                                        mutbl: hir::MutMutable }),
-        ty::ClosureKind::FnOnceClosureKind =>
-            closure_ty
+        ty::ClosureKind::FnClosureKind => tcx.mk_ref(region,
```

Comment:
Yeah, I think we should apply the `let = ` line break heuristic to `=>`, although in this particular case I don't think it's too bad - we space out the struct lits because that is what rustfmt would do and we're not up against the max width limit.